avm/res/compute/virtual-machine/modules/nic-configuration.bicep

```
@@ -0,0 +1,152 @@
+param networkInterfaceName string
+param virtualMachineName string
+param location string
+param tags object?
+param enableIPForwarding bool = false
+param enableAcceleratedNetworking bool = false
+param dnsServers array = []
+@description('Optional. Enable telemetry via a Globally Unique Identifier (GUID).')
+param enableTelemetry bool = true
+
+@description('Optional. The network security group (NSG) to attach to the network interface.')
+param networkSecurityGroupResourceId string = ''
+
+param ipConfigurations array
+param lock lockType
+
+@description('Optional. The diagnostic settings of the Network Interface.')
+param diagnosticSettings diagnosticSettingType
+
+@description('Optional. Array of role assignments to create.')
+param roleAssignments roleAssignmentType
+
+module networkInterface_publicIPAddresses 'br/public:avm/res/network/public-ip-address:0.2.1' = [for (ipConfiguration, index) in ipConfigurations: if (contains(ipConfiguration, 'pipconfiguration')) {
+  name: '${deployment().name}-publicIP-${index}'
+  params: {
+    name: '${virtualMachineName}${ipConfiguration.pipconfiguration.publicIpNameSuffix}'
+    diagnosticSettings: ipConfiguration.?diagnosticSettings
+    location: location
+    lock: lock
+    publicIPAddressVersion: contains(ipConfiguration, 'publicIPAddressVersion') ? ipConfiguration.publicIPAddressVersion : 'IPv4'
+    publicIPAllocationMethod: contains(ipConfiguration, 'publicIPAllocationMethod') ? ipConfiguration.publicIPAllocationMethod : 'Static'
+    publicIpPrefixResourceId: contains(ipConfiguration, 'publicIPPrefixResourceId') ? ipConfiguration.publicIPPrefixResourceId : ''
+    roleAssignments: contains(ipConfiguration, 'roleAssignments') ? ipConfiguration.roleAssignments : []
+    skuName: contains(ipConfiguration, 'skuName') ? ipConfiguration.skuName : 'Standard'
+    skuTier: contains(ipConfiguration, 'skuTier') ? ipConfiguration.skuTier : 'Regional'
+    tags: ipConfiguration.?tags ?? tags
+    zones: contains(ipConfiguration, 'zones') ? ipConfiguration.zones : [
+      '1'
+      '2'
+      '3'
+    ]
+    enableTelemetry: enableTelemetry
```

As per previous comment, should be:
```suggestion
    enableTelemetry: ipConfiguration.?enableTelemetry ?? enableTelemetry
```